plugins/MultiSites/API.php

```diff
@@ -313,6 +313,10 @@ function ($value) {
             );
         }
 
+        // Remove <ts_archived> row metadata, it's already been used by any filters that needed it
+        $dataTable->deleteRowsMetadata(DataTable::ARCHIVED_DATE_METADATA_NAME);
+        $dataTable->deleteColumn('_metadata');
```

Comment:
Was wondering why the percentages were changing constantly. Debugging the code for a while, showed that the metadata removal is actually done to early here. The computed metrics (like evolution) are applied in the post processor, so after the stuff done in API directly. If you want to do that after computing the metrics you need to apply that using a queue filter, as that is executed in a later step. See
https://github.com/matomo-org/matomo/blob/69c09b4c12b22ef5578a335c487eee7bdf3eb312/core/API/DataTablePostProcessor.php#L108-L138

```suggestion
        $dataTable->queueFilter(function($dataTable) {
            $dataTable->deleteRowsMetadata(DataTable::ARCHIVED_DATE_METADATA_NAME);
            $dataTable->deleteColumn('_metadata');
        });
```

Reply:
Good spot! Nice that queueFilter can take a function for simple tasks rather than having to create an actual filter class.